soda-snowflake/src/soda_snowflake/common/data_sources/snowflake_data_source_connection.py

```diff
@@ -36,12 +38,43 @@ class SnowflakePasswordAuth(SnowflakeSharedConnectionProperties):
 
 
 class SnowflakeKeyPairAuth(SnowflakeSharedConnectionProperties):
+    private_key: str = Field(..., description="Private key for authentication")
+    private_key_passphrase: Optional[SecretStr] = Field(None, description="Passphrase if private key is encrypted")
+
+    def to_connection_kwargs(self) -> dict:
+        return {"private_key": self._decrypt(self.private_key, self.private_key_passphrase)}
```

Comment:
Missing the call to parent to get the shared properties here.


```suggestion
        connection_kwargs = super().to_connection_kwargs()
        connection_kwargs["private_key"] = self._decrypt(self.private_key, self.private_key_passphrase)
        return connection_kwargs
```